src/BuildingRegistry.Api.BackOffice/BuildingUnit/BuildingUnitController-Plan.cs

```diff
@@ -70,7 +70,7 @@ public partial class BuildingUnitController
             {
                 throw exception switch
                 {
-                    BuildingUnitCannotBePlannedException =>
+                    BuildingHasInvalidStatusException =>
```

Comment:
I find this exception too common. We added together this more specific exception: BuildingStatusPreventsBuildingUnitRealizationException. The change is not consistent with not realize & realize exceptions.

Reply:
I agree with the inconsistency. We might have to review our exceptions to see if they are not too fine grained.
For instance `BuildingStatusPreventsBuildingUnitRealizationException`: if you do an action Realize Building Unit and you get `BuildingHasInvalidStatusException` does the latter part of the exception matter, cause you know you did realize building.

Also `BuildingHasInvalidStatusException` is thrown from a path where you don't exactly know what was done, could be planned, could be realized, maybe something else.